qir/qat/TestTools/Tests/Unit/vm.cpp

```diff
@@ -0,0 +1,48 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+#include "gtest/gtest.h"
+#include "qir/qat/TestTools/TestVm.hpp"
+
+using namespace microsoft::quantum;
+
+int8_t print_call_value = 0;
```

Comment:
(nit)
The use of a global var. (Theoretically) If multiple tests use it, and they are executed in parallel (in different threads), the behavior can be unexpected.
Consider using a local var instead and passing a pointer or a ref to it as an arg to the funcs that use it, and capturing it by ref in the lambdas that use it.

Reply:
I agree that use of global variables should be discouraged. However, inside a test I think it does very little harm and  using a local or arg does not work in this case.. Re. thread-safety, making the variable local would not solve the issue. Instead you would need to either make it atomic or use  a mutex to lock it. I think the real danger is linking errors due to replicated symbols across translation units. To circumvent this, I've added an anonymous namespace around it.